Project.vcxproj

```diff
@@ -38,29 +38,14 @@
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
 
-  <!-- Okay here's the relevant part -->
-
-  <!-- Step 1: Load SharedFiles.txt in as a string -->
-  <PropertyGroup>
-    <CmakeSharedFilenames_raw>$([System.IO.File]::ReadAllText('$(ProjectDir)SharedFiles.txt').TrimStart().TrimEnd())</CmakeSharedFilenames_raw>
-  </PropertyGroup>
-
-  <!-- Step 2 (optional): Perform a path transformation on each item, in this case prepending src/ -->
-  <PropertyGroup>
-    <CmakeSharedFilenames_raw_pathfix>$([System.Text.RegularExpressions.Regex]::Replace("$(CmakeSharedFilenames_raw)", "^", "src\", "System.Text.RegularExpressions.RegexOptions.Multiline"))</CmakeSharedFilenames_raw_pathfix>
-  </PropertyGroup>
-
-  <!-- Step 3: Convert from a string to a list -->
-  <!-- If you skip step 2, change CmakeSharedFilenames_raw_pathfix here to CmakeSharedFilenames_raw -->
-  <PropertyGroup>
-    <CmakeSharedFilenames>$([System.Text.RegularExpressions.Regex]::Split("$(CmakeSharedFilenames_raw_pathfix)", "\r?\n", "System.Text.RegularExpressions.RegexOptions.Singleline"))</CmakeSharedFilenames>
-  </PropertyGroup>
-
-  <!-- Step 4: Compile the list -->
   <ItemGroup>
+    <!-- Okay here's the relevant part -->
+
     <!-- Shared files -->
-    <!-- To add files to this list, edit SharedFiles.txt -->
-    <ClCompile Include="$(CmakeSharedFilenames)" />
+    <CmakeSharedFilenames Include="$([System.IO.File]::ReadAllText('$(ProjectDir)SharedFiles.txt').Trim().Split($([System.Environment]::NewLine), StringSplitOptions.RemoveEmptyEntries))" />
```

Comment:
Your regex approach with `\r?\n` is more flexible: this will only work with Windows line endings.

Reply:
I'll probably stick with the original then since the goal here is to be cross platform (although I'll leave this PR open in case it's useful to someone)

One question though: If you look you'll see my original has one additional feature not in your oneliner, to wit, it performs a "map" on the input lines (prepends `src/` to each; in the original context this build code was written for, this transformation step was necessary because the CMakeLists and the vcxproj were in different directories and so they prepended different paths). Could this be easily fit into your oneliner version?

Reply:
> the goal here is to be cross platform

Absolutely! Today `.vcxproj` works only on Windows so I'd personally be comfortable with this restriction--that doesn't mean you are.

I would say though that the restriction is more like "you must enable `autocrlf` for `SharedFiles.txt`" than "this works on Windows only"--it's "the line endings in the file on disk must match what .NET thinks is the standard for the current OS".

> it performs a "map" on the input lines (prepends `src/` to each; in the original context this build code was written for, this transformation step was necessary because the CMakeLists and the vcxproj were in different directories and so they prepended different paths). Could this be easily fit into your oneliner version?

I did preserve this but it didn't fit into the oneliner.